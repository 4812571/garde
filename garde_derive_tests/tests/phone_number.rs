#[path = "./util/mod.rs"]
mod util;

#[derive(Debug, garde::Validate)]
struct Test<'a> {
    #[garde(phone_number)]
    field: &'a str,
}

#[test]
fn phone_number_valid() {
    util::check_ok(
        &[
            Test {
                field: "+1 (415) 237-0800",
            },
            Test {
                field: "+14152370800",
            },
            Test {
                field: "+33642926829",
            },
        ],
        &(),
    )
}

#[test]
fn phone_number_invalid() {
    util::check_fail!(
        &[
            Test {
                field: "14152370800"
            },
            Test {
                field: "0642926829"
            },
            Test {
                field: "00642926829"
            },
            Test { field: "A012" },
            Test { field: "TEXT" },
        ],
        &()
    )
}
