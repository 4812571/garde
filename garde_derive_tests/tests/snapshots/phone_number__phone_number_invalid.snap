---
source: garde_derive_tests/tests/phone_number.rs
expression: snapshot
---
Test {
    field: "14152370800",
}
`field` is not a valid phone number: invalid country code

Test {
    field: "0642926829",
}
`field` is not a valid phone number: invalid country code

Test {
    field: "00642926829",
}
`field` is not a valid phone number: invalid country code

Test {
    field: "A012",
}
`field` is not a valid phone number: invalid country code

Test {
    field: "TEXT",
}
`field` is not a valid phone number: not a number


